api/spec/src/billing/invoices.tsp

```diff
@@ -0,0 +1,397 @@
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+namespace OpenMeter.Billing;
+
+/**
+ * Invoices is a collection of endpoints that allow invoice operations without prior
+ * knowledge of the customer ID.
+ */
+@route("/api/v1/billing/invoices")
+@tag("Billing (Experimental)")
+interface InvoiceEndpoints {
+  /**
+   * List invoices
+   */
+  @get
+  @summary("List invoices")
+  @operationId("billingListInvoices")
+  list(
+    @query
+    @summary("Filter by the customer ID or key")
+    customers?: Array<ULIDOrKey>,
+
+    ...InvoiceListParams,
+    ...OpenMeter.QueryPagination,
+    ...OpenMeter.QueryLimitOffset,
+    ...OpenMeter.QueryOrdering<InvoiceOrderBy>,
+  ): OpenMeter.PaginatedResponse<Invoices.Invoice> | OpenMeter.CommonErrors;
+}
+
+/**
+ * CustomerInvoices is a collection of endpoints that allow invoice operations without prior
+ * knowledge of the invoice ID.
+ */
+@route("/api/v1/billing/invoices/{customerIdOrKey}")
+@tag("Billing (Experimental)")
+interface CustomerInvoicesEndpoints {
+  /**
+   * Create a new invoice from the pending items.
+   *
+   * This should be only called if for some reason we need to invoice a customer outside of the normal billing cycle.
+   *
+   * When creating an invoice, the pending items will be marked as invoiced and the invoice will be created with the total amount of the pending items.
+   *
+   * New pending items will be created for the period between now() and the next billing cycle's begining date for any metered item.
+   */
+  @post
+  @summary("Create an invoice")
+  @operationId("billingCreateInvoice")
+  create(
+    @path
+    customerIdOrKey: ULIDOrKey,
+
+    @body
+    request: InvoiceCreateInput,
+  ): {
+    @statusCode _: 201;
+    @body body: Invoices.Invoice;
+  } | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
```

Comment:
cannot be 404

Reply:
If the customer doesn't exists it can be.